pkg/virt-controller/watch/preset.go

```diff
@@ -172,6 +179,55 @@ func (c *VirtualMachinePresetController) initializeVirtualMachine(vmi *kubev1.Vi
 	return nil
 }
 
+func applyNamespaceLimitRangeValues(vmi *kubev1.VirtualMachineInstance, limitrangeInformer cache.SharedIndexInformer) {
+	isMemoryFieldExist := func(resource k8sv1.ResourceList) bool {
+		_, ok := resource[k8sv1.ResourceMemory]
+		return ok
+	}
+
+	vmiResources := vmi.Spec.Domain.Resources
+
+	// Copy namespace memory limits (if exists) to the VM spec
+	if vmiResources.Limits == nil || !isMemoryFieldExist(vmiResources.Limits) {
+
+		namespaceMemLimit, err := getNamespaceLimits(vmi.Namespace, limitrangeInformer)
+		if err == nil {
+			if vmiResources.Limits == nil {
+				vmi.Spec.Domain.Resources.Limits = make(k8sv1.ResourceList)
+			}
+			vmi.Spec.Domain.Resources.Limits[k8sv1.ResourceMemory] = *namespaceMemLimit
+		}
+	}
+
+}
+
+func getNamespaceLimits(namespace string, limitrangeInformer cache.SharedIndexInformer) (*resource.Quantity, error) {
+	finalLimit := &resource.Quantity{Format: resource.BinarySI}
+
+	objListKeys := limitrangeInformer.GetStore().ListKeys()
+	// there can be multiple LimitRange values set for the same resource in
+	// a namespace, we need to find the minimal
+	for _, key := range objListKeys {
+		if strings.HasPrefix(key, namespace) {
+			item, _, _ := limitrangeInformer.GetStore().GetByKey(key)
+			for _, val := range item.(*k8sv1.LimitRange).Spec.Limits {
+				mem := val.Default.Memory()
+				if val.Type == k8sv1.LimitTypeContainer {
```

Comment:
Do we also have the case into account where the limits apply to a `Pod` as a whole?

Reply:
This is another limitation but works differently. Afaiu, it's not possible to express `defaults` on a `Pod` level, only `min`, and `max`. However, I didn't account for this case at all as it's not a resource that we can copy to the VM spec.

Overall, my thinking in this PR was that the namespace limits are applied on the pod but not visible for the user. We then copy it to the VM spec for visibility. But if it fails for another reason, it will be visible for the user. I'm not sure that we want to replicate all the pod validations to the vm, or am I getting it wrong?

Reply:
Probably then also something for #1278 then. :+1: